core/config.go

```diff
@@ -49,14 +72,40 @@ func loadConfigIntoStruct(flags *pflag.FlagSet, target interface{}, configMap *k
 		// Just a single value
 		return key, rawValue
 	})
-	// errors can't occur for this provider
-	_ = configMap.Load(e, nil)
+	return configMap.Load(e, nil)
+}
 
-	// errors can't occur for this provider
-	_ = configMap.Load(posflag.Provider(flags, defaultDelimiter, configMap), nil)
+// loadDefaultsFromFlagset loads the default values, set in the flags, into the configMap.
+// Note: This method should be used first to seed the configMap so other providers can override/alter the configMap.
+func loadDefaultsFromFlagset(configMap *koanf.Koanf, flags *pflag.FlagSet) error {
+	return configMap.Load(posflag.Provider(flags, defaultDelimiter, configMap), nil)
+}
 
-	// load into struct
-	return configMap.UnmarshalWithConf("", target, koanf.UnmarshalConf{
-		FlatPaths: false,
-	})
+// loadFromFlagSet loads the config values set in the command line options into the configMap.
+func loadFromFlagSet(configMap *koanf.Koanf, flags *pflag.FlagSet) error {
+	return configMap.Load(posflag.Provider(flags, defaultDelimiter, configMap), nil)
+}
+
+// LoadConfigMap loads all the values for a given command into the provided configMap.
+// It loads the default values and then (when defined) overwrites them respectively with values from
+// the config file, environment variables and command line flags.
+func LoadConfigMap(configMap *koanf.Koanf, cmd *cobra.Command) error {
+	flags := cmd.Flags()
```

Comment:
do you know what the difference is between persistent flags and non-persistent flags? (otherwise you could just supply the actual flags here, instead of the complete command)

Reply:
Yes, fixed it: server does not have any persistent flags, only the client commands share a common set. Since client commands do not load the config file, the loading code for server and client config is different now. So you no longer need to provide the command for loading the server config, only the flags.